sim_volumes.py

```diff
@@ -0,0 +1,172 @@
+    array
+        list of simulated quaternions of shape [num_pts,4].
+
+    """
+    u = np.random.rand(3, num_pts)
+    u1, u2, u3 = [u[x] for x in range(3)]
+
+    quat = np.zeros((4, num_pts))
+    quat[0] = np.sqrt(1 - u1) * np.sin(np.pi * u2 * 2)
+    quat[1] = np.sqrt(1 - u1) * np.cos(np.pi * u2 * 2)
+    quat[2] = np.sqrt(u1) * np.sin(np.pi * u3 * 2)
+    quat[3] = np.sqrt(u1) * np.cos(np.pi * u3 * 2)
+
+    return np.transpose(quat)
+
+
+def uniform_rotations(num_pts):
+    """Generate ratation as quaternion and convert them as rotation matrix.
+
+    Parameters
+    ----------
+    num : int
+        number of quaternion to return.
+
+    Returns
+    -------
+    rots : ndarray
+        matrix of rotation of the simulated quaternions of shape (num_pts,3,3)
+    qs : array
+        list of simulated quaternions of shape [num_pts,4].
+
+    """
+    qs = get_random_quat(num_pts)
+    rots = SO3.matrix_from_quaternion(qs)  # num,3,3
+    return (rots, qs)
+
+
+def modify_weight(points, volume, vol_size, center):
+    """
+    Fill an empty volume with particles.
+
+    Parameters
+    ----------
+    points : array
+        list particules position.
+    volume : ndarray
+        volume of the molecule.
+    size : int
+        size of the volume.
+    center : int
+        center the molecule around zero.
+
+    Returns
+    -------
+    volume : ndarray
+        volume of the molecule.
+
+    """
+    for point in points.T:
+        for i in range(vol_size):
+            for j in range(vol_size):
+                for k in range(vol_size):
+                    volume[i][j][k] += np.exp(
+                        -np.linalg.norm(
+                            [
+                                i / center - vol_size / center / 2,
+                                j / center - vol_size / center / 2,
+                                k / center - vol_size / center / 2,
+                            ]
+                            - point
+                        )
+                        ** 2
+                        / 2
+                    )
+    return volume
+
+
+def simulate_volumes(particules, n_volumes, vol_size, center=2):
+    """
+    Update a volume with new particles.
+
+    Parameters
+    ----------
+    particules : array
+        list particules position.
+    n_volumes : int
+        number of data.
+    img_size : int
+        size of the molecule.
+    center : int
+        optional center the molecule around zero.
+
+    Returns
+    -------
+    volumes : ndarray
+        3D map of the molecule.
+    qs : list
+        describes rotations in quaternions.
+
+    """
+    rots, qs = uniform_rotations(n_volumes)
+    volumes = np.zeros((n_volumes,) + (vol_size,) * 3)
+    for idx in range(n_volumes):
+        if idx % (n_volumes / 10) == 0:
+            print(idx)
+        points = rots[idx].dot(particules)
+        volumes[idx] = modify_weight(points, volumes[idx], vol_size, center)
+    qs = [np.array2string(q) for q in qs]
```

Comment:
Why do we want to convert to string? it might not seem natural from a user perspective that this function returns a string of a quaternion. Is it possible to just return the quaternion, and convert it to string whenever it's really needed?